### DISCUSSION TITLE
Different Executors for Different Transports

### DISCUSSION THREAD

**Discussion by MavenRain** [libp2p/rust-libp2p#2565] [Q&A] [answered]
What I am discovering in our company's use of libp2p is that different executors for swarms work with different transports.  When I use the `MemoryTransport`, the default executor will work, and I can create a swarm just using `Swarm::new`.  If I use a transport constructed from `TokioTcpConfig`, then I must implement an (albeit simple) executor manually, to be added via `SwarmBuilder::new`.  Is this by design?

**Comment by mxinden** [comment, marked as answer]
You are correct. When using `TokioTcpConfig` you need to provide an executor. See also https://github.com/libp2p/rust-libp2p/issues/2230.

I am sorry you had to debug this yourself. To prevent others from hitting this issue as well, would you mind sending in a patch to the documentation?

Also in case you can think of a way to prevent this at compile time, comments in https://github.com/libp2p/rust-libp2p/issues/2230 would be appreciated.

**Reply by MavenRain** [reply]
@mxinden I can look at making an edit to the documentation to draw attention to this, and will see about a compile-time solution.
